### PR TITLE
Increase module performance, reduce memory and disk footprint

### PR DESCRIPTION
**Pull request by martijnarts**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,10 +1,7 @@
-var VOWELS   = /[aeiou]/gi;
-var VOWELS_AND_SPACE = /[aeiou\s]/g;
-
 exports.parse = function parse(string, join){
   if(typeof string !== 'string'){
     throw new TypeError('Expected a string as the first option');
   }
 
-  return string.replace(join ? VOWELS_AND_SPACE : VOWELS, '');
+  return string.replace(join ? /[aeiou\s]/gi : /[aeiou]/gi);
```
Comment (lieuwex) [inline index.js]: Missing the second parameter here, running this will replace all vowels with 'undefined'.

**Comment by stephenswat** [comment]
Why is this even a module at this point? Surely adding a bunch of dependencies and overhead to your program for _one_ non-flexible single-line string transformation that has almost no practical use cases would sound like a _bad idea_ to anyone but the biggest node.js fetishists who cower in fear at the thought of having to do any actual programming?

File: index.js (hunk 1 above)
Comment (martijnarts) [inline index.js]: Tests were broken :+1:

**Comment by martijnarts** [comment]
@StephenSwat you have obviously not been keeping up. There's been a lot of [discussion](http://www.reddit.com/r/programming/comments/4bjss2/_/) recently on [this exact topic](https://github.com/sindresorhus/ama/issues/10#issuecomment-117766328), mostly fuelled by [a few modules](https://medium.com/@azerbike/i-ve-just-liberated-my-modules-9045c06be67c) that were removed from the NPM repository.

**Comment by lestoni** [comment]
@TotempaaltJ Thanks man, looks neat :+1: I wrote this module while learning Nodejs, quiet naive beginnings i might say. Glad it found some use though.

**Comment by lestoni** [comment]
@StephenSwat Yes,i do understand your concern but there is no need to overthink about it.

**Comment by stephenswat** [comment]
@lestoni please note that I'm not criticizing you. I think it's fantastic that you're contributing (to) free software and I think you should continue because that's a _great_ thing to do. My rant is directed only at this JavaScript idea that everything should be a module (partially because the JavaScript standard library is.. lacking).

**Pull request merged**